### PR TITLE
Update seasonal cycle analysis

### LINKED ISSUE #15: Improve seasonal cycle evaluation
At the moment the seasonal cycle results in `results/seasonal_cycle.ipynb` don't tell us much because they represent the average cycle over the whole country (`seas_cycle = climatology.mean(dim=["lat", "lon"])`).

I've started playing around with alternative ways to look at the seasonal cycle at point locations (see https://github.com/DamienIrving/npcp/blob/master/results/monthly_climatology.ipynb). I've still got a bit of work to do, but if you look at Brisbane during May or Alice Springs during June you can see that the bias correction produces some crazy results.

I need to look at the bias correction process in more detail to understand why those crazy values happen... here's a list of things to look at:
- [ ] The sanity check notebook at `/g/data/ia39/npcp/data/pr/CSIRO-ACCESS-ESM1-5/BOM-BARPA-R/ecdfm/task-historical`
- [ ] The problem might be somewhat related to the short training period we're using (20 years). I've plotted the seasonal cycle with 20, 30 and 40 year time periods to try and visualise what difference is: https://github.com/climate-innovation-hub/qq-workflows/blob/main/npcp/monthly_climatology.ipynb

I should also update the notebook so that it uses monthly AGCD data too - see #14

### PR DESCRIPTION
**Pull request by DamienIrving**
Resolves #15

### CODE REVIEW

**Comment by AliciaTak** [comment]
@DamienIrving It looks good. My only suggestion is to reduce the line widths in the line plots to be able to distinguish the overlapping lines better. As an alternative (you don't have to do it), you could use a metric like [Fréchet distance](https://pypi.org/project/frechetdist/ ) to measure the similarities between AGCD and ECDFm curves, and AGCD and QME curves. But it might be too much work, so I'll leave it to you to decide.
A Question - you suggested to me to calculate the root mean squared bias for the 5year max/min metric, why don't you do that here as well instead of the spatial average bias?

**Comment by DamienIrving** [comment]
Thanks, @AliciaTak. I'll take a look at the Frechet distance and see whether it's worth investing the time.

Good question regarding the RMSE - in these notebooks I've calculated the mean absolute error (MAE; although in this case all values of my error metric are positive so taking the absolute value isn't relevant) rather than the RMSE. I guess I like the MAE better because every error has equal weighting, whereas larger errors have a disproportionately large impact on the RMSE score: https://en.wikipedia.org/wiki/Mean_absolute_error

Maybe we should decide on either the RMSE or MAE and use it consistently for our spatial plots?

**Comment by AliciaTak** [comment]
> Thanks, @AliciaTak. I'll take a look at the Frechet distance and see whether it's worth investing the time.
> 
> Good question regarding the RMSE - in these notebooks I've calculated the mean absolute error (MAE; although in this case all values of my error metric are positive so taking the absolute value isn't relevant) rather than the RMSE. I guess I like the MAE better because every error has equal weighting, whereas larger errors have a disproportionately large impact on the RMSE score: https://en.wikipedia.org/wiki/Mean_absolute_error
> 
> Maybe we should decide on either the RMSE or MAE and use it consistently for our spatial plots?

@DamienIrving sounds good, I am happy with either.

File: results/utils.py [hunk 1]
```diff
@@ -96,9 +96,12 @@ def get_npcp_data(
...
     if region == 'AU':
         shape = get_aus_shape()
-    else:
+        da = clip_data(ds[variable], shape)
+    elif region:
         shape = get_nrm_super_cluster(region)
-    da = clip_data(ds[variable], shape)
+        da = clip_data(ds[variable], shape)
+    else:
+        da = ds[variable]
```
Comment (DamienIrving) [inline results/utils.py:105]: @AliciaTak - We were clipping the data twice (once when we first got the data and again for the plotting) so I've edited the `get_npcp_data` function so the default is to not clip at all. That should help things run a little more efficiently.

**Comment by AliciaTak** [comment]
> @AliciaTak I've updated these notebooks to run a little more efficiently and I've also made the lines thinner like you suggested. If we merge this one you'll be able to use the same efficiencies to `utils.py`.

@DamienIrving Thanks, Damien, it looks good, you can merge! So, are we using MAE from now on?

**Comment by DamienIrving** [comment]
@AliciaTak - Yeah, I think MAE. I'll update the mean bias notebooks :smile:

**Pull request merged**